src/components/compound/DlSearches/DlSmartSearch/DlSmartSearch.vue

```diff
@@ -61,16 +62,73 @@
                 </dl-menu>
             </dl-button>
         </div>
-        <dl-json-editor
+        <dl-dialog-box
             :model-value="jsonEditorModel"
-            :query="activeQuery"
-            :queries="filters.saved"
-            @update:modelValue="jsonEditorModel = $event"
-            @save="saveQueryDialogBoxModel = true"
-            @remove="handleQueryRemove"
-            @search="handleQuerySearchEditor"
-            @update-query="handleEditorQueryUpdate"
-        />
+            :width="800"
+        >
+            <template #header>
+                <dl-dialog-box-header
+                    title="DQL Search"
+                    :close-button="true"
+                    style="font-weight: 200"
+                    @onClose="handleJsonEditorClose"
+                />
+            </template>
+            <template #body>
+                <div class="json-editor-layout">
+                    <div class="json-query">
+                        <div class="json-query-menu">
+                            <dl-select
+                                :model-value="selectedOption"
+                                width="200px"
+                                :options="selectOptions"
+                                placeholder="New Query"
+                                @update:model-value="updateActiveQuery"
+                            />
+                            <dl-button
```

Comment:
if this button is for formatting then it should be part of the dl-jsosn-editor component.

Reply:
by design this should be aligned with with the Select, so it would be a bit weird to have these seemingly  at the same level in the same container but in different components. The code for alignment boils down to a JSON.stringify with indentation too, so I thought it's ok to just change the string from outside like that

Reply:
I understand maybe for the future we can add a slot there in the json editor for more action buttons.